### PR TITLE
Implement basic invite logic with database

### PR DESCRIPTION
**Pull request by jonkoops**

### CODE REVIEW

File: src/db.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    pub email: Option<String>,
+}
+
+pub enum InvitationStatus {
+    Pending,
+    Accepted,
+    Rejected,
+}
+
+impl From<String> for InvitationStatus {
```
Comment (samuelleeuwenburg) [inline src/db.rs:18]: Maybe use `TryFrom` instead and don’t panic

File: src/db.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+pub async fn get_invitation_by_code(pool: &SqlitePool, code: &str) -> Option<Invitation> {
+    sqlx::query_as!(
+        Invitation,
+        r#"SELECT id, status, code, email FROM invitations WHERE code = ?"#,
+        code
+    )
+    .fetch_optional(pool)
+    .await
+    .unwrap()
```
Comment (samuelleeuwenburg) [inline src/db.rs:56]: Don’t unwrap here but let the caller handle the Result as wel as the Option

File: src/db.rs (hunk 1 above)
Comment (jonkoops) [reply src/db.rs:18]: Tried this, but SQLx doesn't implement `TryFrom`, but only `From`, so I'll have to leave it as-is. It's not a public trait, so likely this is fine for now.

File: src/db.rs (hunk 2 above)
Comment (jonkoops) [reply src/db.rs:56]: Tried doing that, but then the return type becomes really complex.

**Pull request merged**